sgx-types/src/secs.rs

```diff
@@ -27,34 +27,48 @@ impl Spec {
     ///
     /// # Safety
     /// This function is unsafe because it does not check if the `CPUID` instruction
-    /// is available before issuing it.
-    pub unsafe fn max_enc_size() -> Option<NonZeroU64> {
+    /// is available before issuing it, which in rare cases could result in a crash on
+    /// unsupported CPUs. According to common practice, these calls
+    /// are wrapped in an unsafe block without making the whole function unsafe. For
+    /// more, see https://github.com/rust-lang/rust/issues/60123.
+    pub fn max_enc_size() -> Option<NonZeroU64> {
         const LEAF_MAX_PARAM: u32 = 0x0;
         const LEAF_SGX_SUPPORT: u32 = 0x07;
         const SUBLEAF_SGX_SUPPORT: u32 = 0x0;
         const LEAF_MAX_ENCL_SIZE: u32 = 0x12;
         const SUBLEAF_MAX_ENCL_SIZE: u32 = 0x0;
 
         // Test for max leaf size
-        let res = core::arch::x86_64::__get_cpuid_max(LEAF_MAX_PARAM);
+        let res = unsafe { core::arch::x86_64::__get_cpuid_max(LEAF_MAX_PARAM) };
         let max_leaf = res.0;
         if max_leaf < LEAF_SGX_SUPPORT || max_leaf < LEAF_MAX_ENCL_SIZE {
             return None;
         }
 
         // Test for SGX support
-        let res = core::arch::x86_64::__cpuid_count(LEAF_SGX_SUPPORT, SUBLEAF_SGX_SUPPORT);
+        let res =
+            unsafe { core::arch::x86_64::__cpuid_count(LEAF_SGX_SUPPORT, SUBLEAF_SGX_SUPPORT) };
```

Comment:
Rather than wrapping the line here, do `use core::arch::x86_64::__cpuid_count;` at the top of the function.